tests/adapter/dbt/tests/adapter/utils/data_types/test_type_boolean.py

```diff
@@ -0,0 +1,24 @@
+import pytest
+from dbt.tests.adapter.utils.data_types.base_data_type_macro import BaseDataTypeMacro
+
+seeds__expected_csv = """boolean_col
+True
+""".lstrip()
+
+models__actual_sql = """
+select cast('True' as {{ type_boolean() }}) as boolean_col
+"""
```

Comment:
Probably want to cover both boolean options at the very least.
```suggestion
seeds__expected_csv = """boolean_col
True
False
""".lstrip()

models__actual_sql = """
select cast('True' as {{ type_boolean() }}) as boolean_col
union all
select cast('False' as {{ type_boolean() }}) as boolean_col
"""
```

### Even better
But test cases covering the truth tables for [conjunction, disjunction, and negation](https://en.wikipedia.org/wiki/Boolean_algebra#Basic_operations) would be even better. That way, we are verifying that everything is _acting_ like booleans.

<img width="120" alt="image" src="https://user-images.githubusercontent.com/44704949/191152212-e690222c-3a3d-4e2e-acef-51208c5d4407.png"> <img width="120" alt="image" src="https://user-images.githubusercontent.com/44704949/191152234-db9ea5a1-b28a-4b5c-aa51-6dd049555661.png"> <img width="120" alt="image" src="https://user-images.githubusercontent.com/44704949/191152253-41b1b12a-6ded-47d7-bae6-2e24be29d2b4.png">

Something like the following **untested** code:
```
seeds__boolean_permutations_csv = """
x,y
False,False
True,False
False,True
True,True
""".lstrip()

seeds__expected_csv = """
x,y,conjunction,disjunction,negation_x
False,False,False,False,True
True,False,False,True,False
False,True,False,True,True
True,True,True,True,False
""".lstrip()

models__actual_sql = """
select
    x,
    y,
    x and y as conjunction,
    x or y as disjunction,
    not x as negation_x
from {{ ref("boolean_permutations" }}
```

### Best? 🤷 

Even though `BOOLEAN` is a data type described in the SQL standard, some databases don't have it (looking at you, SQL Server!). 

If we are feeling extra magnanimous, we could change all the True/False values in the seeds to be `1`/`0` instead.

I'm hoping it wouldn't be necessary, but we could update the `models__actual_sql` definition so that x/y values are replaced with the following instead:
- `cast(x as {{ type_boolean() }})`
- `cast(y as {{ type_boolean() }})`

Reply:
Update:

The "Even better" example I gave above didn't actual test the `type_boolean()` macro at all! 😅 

Something like this should fix that situation:
```
models__actual_sql = """
select
    cast(x as {{ type_boolean() }}) as x_bool,
    cast(y as {{ type_boolean() }}) as y_bool,
    cast(x as {{ type_boolean() }}) and cast(y as {{ type_boolean() }}) as conjunction,
    cast(x as {{ type_boolean() }}) or cast(y as {{ type_boolean() }}) as disjunction,
    not cast(x as {{ type_boolean() }}) as negation_x
from {{ ref("boolean_permutations" }}
```

Reply:
Cheers @dbeatty10. I'll take a second pass at this later today :)